src/keys.cpp

```diff
@@ -463,7 +463,11 @@ long PubKey::Encrypt(Ctxt& ctxt,
       NTL::conv<double>(embeddingLargestCoeff(ptxt_fixed, context.zMStar));
 
   if (ptxt_sz > ptxt_bound) {
+#ifdef HELIB_DEBUG
     Warning("noise bound exceeded in encryption");
+#else
+    throw LogicError("noise bound exceeded in encryption");
+#endif
```

Comment:
```suggestion
    // TODO: Turn the following preprocessor logics into a warnOrThrow function
#ifdef HELIB_DEBUG
    Warning("noise bound exceeded in encryption");
#else
    throw LogicError("noise bound exceeded in encryption");
#endif
```
Turn the following preprocessor logics into a warnOrThrow function